framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1079,6 +1084,43 @@ protected void printOrStoreMessage(
         Trees.instance(processingEnv).printMessage(kind, message, source, root);
     }
 
+    /**
+     * printOrStoreMessage method with an added stack trace argument. The stack trace is printed
+     * when the dumpOnErrors option is enabled.
+     *
+     * @param kind the kind of message to print
+     * @param message the message text
+     * @param source the souce code position of the diagnostic message
+     * @param root the compilation unit
+     * @param trace the stack trace where the checker encountered an error
+     */
+    protected void printOrStoreMessage(
+            javax.tools.Diagnostic.Kind kind,
+            String message,
+            Tree source,
+            CompilationUnitTree root,
+            StackTraceElement[] trace) {
+        Trees.instance(processingEnv).printMessage(kind, message, source, root);
+        printStackTrace(trace);
+    }
+
+    /**
+     * Output the given stack trace if the "dumpOnErrors" option is enabled.
+     *
+     * @param trace stack trace when the checker encountered a warning/error
+     */
+    private void printStackTrace(StackTraceElement[] trace) {
+        boolean dumpOnErrors =
+                getOptions().containsKey("dumpOnErrors") && getBooleanOption("dumpOnErrors", true);
```

Comment:
Shouldn't this just be `getBooleanOption("dumpOnErrors")`? That method already checks whether the option is given and false should be the default, shouldn't it?

Reply:
I originally tried ```getBooleanOption("dumpOnErrors", false)```, however it returns false whenever no boolean argument is given. ```getBooleanOption("dumpOnErrors", true)``` works for the most part but returns true even if the ```dumpOnErrors``` option is not provided. To prevent that, I used the ```getOptions().containsKey("dumpOnErrors")``` to check whether the dumpOnErrors option is given.

Reply:
That's a bit odd, looking at https://github.com/typetools/checker-framework/blob/34989702ef50f1ae3939d7acda3ad3b8dabc56e1/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java#L1589 I wouldn't expect that behavior.
Can you investigate why this doesn't work as specified?

Reply:
If we write ```dumpOnErrors``` without a boolean argument, the ```value``` variable in the ```getBooleanOption()``` method would be null. Hence, the function would return the default value which, if we used ```getBooleanOption("dumpOnErrors", false)``` would be false. So the stack trace won't be printed if we don't give the boolean value ```true```. If we use ```getBooleanOption("dumpOnErrors", true)```, then we don't need a boolean argument to print the stack trace, however, then it would print the stack trace even if ```dumpOnErrors``` is not provided because ```value``` will be null, and it would return the default value which is ```true```.

Reply:
Is simply `hasOption("dumpOnErrors")` enough? Then you don't need to introduce a local variable for this.
In a separate PR, can you look into what `getBooleanOption` is doing wrong? It is hardly used, so we should discuss what the correct behavior is.

Reply:
```hasOption("dumpOnErrors")``` is the same as ```getOptions().containtsKey("dumpOnErrors")```. If we give the option ```-AdumpOnErrors=false```, ```hasOption("dumpOnErrors")``` will return true. Hence, we cannot use this directly. I have replaced ```getOptions().containtsKey("dumpOnErrors")``` with ```hasOption("dumpOnErrors")```.